server/gui/main.py

```diff
@@ -155,14 +154,15 @@ def __init__(self, parent):
         self.signals.selectedFile.connect(self.createScanpyEngine)
 
     def updateEmbedding(self, idx):
-        self.embedding_selection = MODES[idx]
+        self.embedding_selection = [MODES[idx]]
 
     def createScanpyEngine(self, file_name):
-        worker = DataLoadWorker(file_name, self.embedding_selection)
-        worker.signals.result.connect(self.onDataSuccess)
-        worker.signals.error.connect(self.onDataError)
-        self.load_worker = threading.Thread(target=worker.run, daemon=True)
-        self.load_worker.start()
+        worker = Worker(self.window().child_conn, file_name, self.title, host="127.0.0.1", port=GUI_PORT,
+                        layout=self.embedding_selection)
```

Comment:
do you want to be restricting to a single layout?   The back-end is no longer responsible for selecting layouts, but rather simply for enumerating them and providing all to the front-end.  See PR #766.

Reply:
For now, yes. I want a UI element besides file selection in the layout. With the intention to fill out the selection UI later (after installer is proved).

Reply:
I don't think we necessarily want to move these UI controls out of the front-end and into the wrapper frame.  This seems like an open design question....  (I'm actually not sure what I think is the right approach, but we need mocks & UX testing on the larger issue)

Reply:
Totally agree, I just want to delay that issue until after I've got installer working but before release.